### PR TITLE
Add a new method "AttachEphemeralDisk" for virtual guest

### PR DESCRIPTION
**Pull request by xingzhou**
this method is used for attach a second local disk to the virtual guest.

Signed-off-by: Edward Zhang <zhuadl@cn.ibm.com>

### CODE REVIEW

File: data_types/softlayer_item_price.go [hunk 1]
```diff
@@ -10,3 +11,7 @@ type Item struct {
 	Description string `json:"description"`
 	Capacity    string `json:"capacity"`
 }
+
+type Category struct {
+	CategoryCode string `json:"categoryCode"`
```
Comment (maximilien) [inline data_types/softlayer_item_price.go]: Should this one use `omitempty`? That is, is it optional?

File: services/softlayer_virtual_guest.go [hunk 2]
```diff
@@ -404,6 +405,46 @@ func (slvgs *softLayer_Virtual_Guest_Service) IsPingable(instanceId int) (bool,
...
+
+	order := datatypes.SoftLayer_Product_Order{
+		VirtualGuests: []datatypes.VirtualGuest{
+			datatypes.VirtualGuest{
+				Id: instanceId,
+			},
+		},
+		Prices: []datatypes.SoftLayer_Item_Price{
+			datatypes.SoftLayer_Item_Price{
+				Id: 29190, // ToDO: now we just attach a 25G local disk, need to query local disk size according to the diskSize parameter
```
Comment (maximilien) [inline services/softlayer_virtual_guest.go]: So I assume you plan to fix this soon?

File: services/softlayer_virtual_guest.go [hunk 3]
```diff
@@ -404,6 +405,46 @@ func (slvgs *softLayer_Virtual_Guest_Service) IsPingable(instanceId int) (bool,
...
+		},
+		ComplexType: "SoftLayer_Container_Product_Order_Virtual_Guest_Upgrade",
+		Properties: []datatypes.Property{
+			datatypes.Property{
+				Name:  "MAINTENANCE_WINDOW",
+				Value: time.Now().UTC().Format(time.RFC3339Nano),
+			},
+			datatypes.Property{
+				Name:  "NOTE_GENERAL",
+				Value: "addingdisks",
```
Comment (maximilien) [inline services/softlayer_virtual_guest.go]: Can this value be parametrized? That is can this be passed in the call somewhere? Or is it fixed?

File: services/softlayer_virtual_guest_test.go [hunk 4]
```diff
@@ -166,6 +166,18 @@ var _ = Describe("SoftLayer_Virtual_Guest_Service", func() {
...
+			fakeClient.DoRawHttpRequestResponse, err = common.ReadJsonTestFixtures("services", "SoftLayer_Product_Order_placeOrder.json")
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("can attach a local disk without error", func() {
+			err := virtualGuestService.AttachEphemeralDisk(123, 25)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
```
Comment (maximilien) [inline services/softlayer_virtual_guest_test.go]: Is there a way to test getting the list of disks and thus verifying that after this call the disk was attached and can be seen?

File: softlayer/softlayer_virtual_guest_service.go [hunk 5]
```diff
@@ -40,4 +40,6 @@ type SoftLayer_Virtual_Guest_Service interface {
 
 	AttachIscsiVolume(instanceId int, volumeId int) (string, error)
 	DetachIscsiVolume(instanceId int, volumeId int) error
```
Comment (maximilien) [inline softlayer/softlayer_virtual_guest_service.go]: I do not see any code for `DetachEphemeralDisk`? Is this coming after?

**Comment by maximilien** [comment]
OK so I am accepting this to move fast, however, please address each of the comments and submit next PR with improvements. Thanks,

dr.max

File: data_types/softlayer_item_price.go (hunk 1 above)
Comment (xingzhou) [inline data_types/softlayer_item_price.go]: Categories field in Softlayer_Item_Price can be empty, but if we have one Category listed, the categorycode must be provided. it's a mandatory field at present.

File: services/softlayer_virtual_guest.go (hunk 2 above)
Comment (xingzhou) [inline services/softlayer_virtual_guest.go]: This has been fixed in the latest pull request #21

File: services/softlayer_virtual_guest.go (hunk 3 above)
Comment (xingzhou) [inline services/softlayer_virtual_guest.go]: This is just a description of what has been upgraded, do you think we need to cusomize the description? if no, we can just keep it as a const

File: services/softlayer_virtual_guest_test.go (hunk 4 above)
Comment (xingzhou) [inline services/softlayer_virtual_guest_test.go]: for unit test, I'm thinking that we are using some fake http request to satisfy our testing scenario, so we suggest not put this testing scenario into the unit test

File: softlayer/softlayer_virtual_guest_service.go (hunk 5 above)
Comment (xingzhou) [inline softlayer/softlayer_virtual_guest_service.go]: We can add detachEphemeralDisk for softlayer-go, can submit this in another independent story. But I'm not seeing if BOSH needs this feature till now?

**Pull request merged**